h2o-core/src/main/java/water/api/ModelBuilderHandler.java

```diff
@@ -38,11 +39,23 @@ public S handle(int version, Route route, Properties parms, String postBody) thr
 
     // User specified key, or make a default?
     String model_id = parms.getProperty("model_id");
+    String warningStr = null;
```

Comment:
Can this be simplified?

You don't need modifiedModelId, you can add warningStr if it is not null. I would also avoid adding a new constructor to Job, this class already has a rich constructor.